packages/@react-aria/table/docs/useTable.mdx

```diff
@@ -488,12 +492,823 @@ function Checkbox(props) {
+              )
+            )}
+          </ResizableTableHeaderRow>
+        ))}
+      </ResizableTableRowGroup>
+      <ResizableTableRowGroup
+        /*- begin highlight -*/
+        style={{
+          maxHeight: '200px'
+        }}
+        /*- end highlight -*/
+        type="tbody">
+        {[...collection.body.childNodes].map(row => (
+          <ResizableTableRow key={row.key} item={row} state={state}>
+            {[...row.childNodes].map(cell =>
+              cell.props.isSelectionCell ? (
+                <ResizableTableCheckboxCell
+                  key={cell.key}
+                  cell={cell}
+                  state={state}
+                  /*- begin highlight -*/
+                  widths={widths}
+                  /*- end highlight -*/
+                />
+              ) : (
+                <ResizableTableCell
+                  key={cell.key}
+                  cell={cell}
+                  state={state}
+                  /*- begin highlight -*/
+                  widths={widths}
+                  /*- end highlight -*/
+                />
+              )
+            )}
+          </ResizableTableRow>
+        ))}
+      </ResizableTableRowGroup>
+    </table>
+  );
+}
+```
+
+</details>
+
+
+### Resizable table header
+
+The `TableRowGroup` and `TableHeaderRow` only require minor style changes to accommodate the the new `display` changes made in
+`ResizableColumnsTable`.
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show code</summary>
+
+```tsx example export=true render=false
+function ResizableTableRowGroup({type: Element, style, children}) {
+  let {rowGroupProps} = useTableRowGroup();
+  return (
+    <Element
+      {...rowGroupProps}
+      style={{
+        /*- begin highlight -*/
+        display: 'block',
+        /*- end highlight -*/
+        ...style
+      }}>
+      {children}
+    </Element>
+  );
+}
+```
+
+</details>
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show code</summary>
+
+```tsx example export=true render=false
+function ResizableTableHeaderRow({item, state, children}) {
+  let ref = useRef();
+  let {rowProps} = useTableHeaderRow({node: item}, state, ref);
+
+  return (
+    <tr
+      {...rowProps}
+      ref={ref}
+      /*- begin highlight -*/
+      style={{display: 'flex'}}
+      /*- end highlight -*/
+    >
+      {children}
+    </tr>
+  );
+}
+```
+
+</details>
+
+The `TableColumnHeader` is where we see the bulk of the changes required to support resizable columns. First of all, we need to accommodate a `Resizer` element in every resizable column that
+the user can drag or focus to perform a resize operation. To avoid interfering with the existing grid keyboard navigation, we only display this resizer when
+the column is hovered or if the column itself is being resized. This however introduces another issue: how will keyboard, screen reader, and touch users begin a resize operation if the resizer
```

Comment:
This is actually a bug (?) in our TableView, if you hover over the column headers and then use left/right to navigate through the columns, you'll actually land on the resizer and enter resizing mode. It also feels like a bit of a strange restriction to require the user to hide the resizer to avoid grid keyboard navigation issues

Did some thinking about this, maybe it would be nice to make an arbitrary element excluded from the FocusableTreewalker, open to ideas

Reply:
IMO that would depend if we'd want to make that treewalker filtering/excluding public api or for internal use only. Maybe it could be a prop/callback like you said, maybe we provide it through a context since useGridCell is a bit deep sometimes, maybe we have a specific DOM attribute we look for and auto filter elements that have that attribute (heh)